satpy/readers/clavrx.py

```diff
@@ -32,183 +37,103 @@
 
-    def get_nadir_resolution(self, sensor):
-        """Get nadir resolution."""
-        for k, v in self.nadir_resolution.items():
-            if sensor.startswith(k):
-                return v
-        res = self.filename_info.get('resolution')
-        if res.endswith('m'):
-            return int(res[:-1])
-        elif res is not None:
-            return int(res)
 
-    @property
-    def start_time(self):
-        """Get the start time."""
-        return self.filename_info['start_time']
+def _get_rows_per_scan(sensor: str) -> str:
+    """Get number of rows per scan."""
+    for k, v in ROWS_PER_SCAN.items():
+        if sensor.startswith(k):
+            return v
+    return None
 
-    @property
-    def end_time(self):
-        """Get the end time."""
-        return self.filename_info.get('end_time', self.start_time)
 
-    def available_datasets(self, configured_datasets=None):
-        """Automatically determine datasets provided by this file."""
-        sensor = self.get_sensor(self['/attr/sensor'])
-        nadir_resolution = self.get_nadir_resolution(sensor)
-        coordinates = ('longitude', 'latitude')
-        handled_variables = set()
+def _get_data(data: xr.DataArray, dataset_id: dict, ds_info: dict) -> xr.DataArray:
+    """Get a dataset."""
+    if dataset_id.get('resolution'):
+        data.attrs['resolution'] = dataset_id['resolution']
 
-        # update previously configured datasets
-        for is_avail, ds_info in (configured_datasets or []):
-            this_res = ds_info.get('resolution')
-            this_coords = ds_info.get('coordinates')
-            # some other file handler knows how to load this
-            if is_avail is not None:
-                yield is_avail, ds_info
+    attrs = data.attrs.copy()
+    fill = attrs.pop('_FillValue', None)
+    factor = attrs.pop('scale_factor', None)
+    offset = attrs.pop('add_offset', None)
+    valid_range = attrs.pop('valid_range', None)
 
-            var_name = ds_info.get('file_key', ds_info['name'])
-            matches = self.file_type_matches(ds_info['file_type'])
-            # we can confidently say that we can provide this dataset and can
-            # provide more info
-            if matches and var_name in self and this_res != nadir_resolution:
-                handled_variables.add(var_name)
-                new_info = ds_info.copy()  # don't mess up the above yielded
-                new_info['resolution'] = nadir_resolution
-                if self._is_polar() and this_coords is None:
-                    new_info['coordinates'] = coordinates
-                yield True, new_info
-            elif is_avail is None:
-                # if we didn't know how to handle this dataset and no one else did
-                # then we should keep it going down the chain
-                yield is_avail, ds_info
+    if factor is not None and offset is not None:
+        def scale_inplace(data):
+            data *= factor
+            data += offset
```

Comment:
I'm not sure I feel strongly enough about this to request a change, but we have to be careful with inplace operations on dask-backed DataArrays. I've had problems in some code where multiplying an integer array by a float (factor) can lead to an error because the result of `int * float` should be a float but the inplace array is `int`. If this doesn't need to be in-place, could you make it not in-place?

Reply:
Yes, I can change that.  I tried to change as little as possible from the previous code, but I am not attached to the scale in place operations.

Reply:
It may not actually be a problem, but with dask there also shouldn't be a point to doing it...so let's go not in place.